files/halite_gem/poise_python/resources/python_package.rb

```diff
@@ -71,20 +90,45 @@ module PythonPackage
     finder_options['process_dependency_links'] = options.process_dependency_links
   if getattr(options, 'format_control', None):
     finder_options['format_control'] = options.format_control
-  finder = PackageFinder(**finder_options)
-  find_all = getattr(finder, 'find_all_candidates', getattr(finder, '_find_all_versions', None))
-  for arg in args:
+  try:
+    finder = PackageFinder(**finder_options)
+  except TypeError:
+    # need to use LinkCollector
     try:
-      from pip._internal.req.constructors import install_req_from_line
-      req = install_req_from_line(arg)
+      from pip._internal.collector import LinkCollector
     except ImportError:
-      from pip._internal.req import InstallRequirement
-      req = InstallRequirement.from_line(arg)
+      from pip._internal.index.collector import LinkCollector
+
+    from pip._internal.models.search_scope import SearchScope
+    from pip._internal.models.selection_prefs import SelectionPreferences
+
+    search_scope = SearchScope.create(
+      finder_options['find_links'], finder_options['index_urls']
+    )
+    collector = LinkCollector(session, search_scope)
+    selection_prefs = SelectionPreferences(
+      False, # allow_yanked
```

Comment:
i don't understand why `allow_yanked` is a comment instead of a kwarg

Reply:
it wasn't listed as one in the source but TIL anything can be a kwarg in python; will make the change.